service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerTest.java

```diff
@@ -0,0 +1,235 @@
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+
+    @Test
+    void aboutToStartCallbackShouldSplitDirectionsIntoSeparateCollections() throws Exception {
+        String title = "example direction";
+
+        List<Direction> directions = ImmutableList.of(
+            Direction.builder().type(title).assignee(ALL_PARTIES).build(),
+            Direction.builder().type(title).assignee(LOCAL_AUTHORITY).build(),
+            Direction.builder().type(title).assignee(PARENTS_AND_RESPONDENTS).build(),
+            Direction.builder().type(title).assignee(CAFCASS).build(),
+            Direction.builder().type(title).assignee(OTHERS).build(),
+            Direction.builder().type(title).assignee(COURT).build()
+        );
+
+        Order sdo = Order.builder().directions(buildDirections(directions)).build();
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of("standardDirectionOrder", sdo))
+                .build())
+            .build();
+
+        MvcResult response = mockMvc
+            .perform(post("/callback/draft-SDO/about-to-start")
+                .header("authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(mapper.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+        assertThat(extractDirections(caseData.getAllParties())).containsOnly(directions.get(0));
+        assertThat(extractDirections(caseData.getLocalAuthorityDirections())).containsOnly(directions.get(1));
+        assertThat(extractDirections(caseData.getParentsAndRespondentsDirections())).containsOnly(directions.get(2));
+        assertThat(extractDirections(caseData.getCafcassDirections())).containsOnly(directions.get(3));
+        assertThat(extractDirections(caseData.getOtherPartiesDirections())).containsOnly(directions.get(4));
+        assertThat(extractDirections(caseData.getCourtDirections())).containsOnly(directions.get(5));
+    }
+
+    @Test
+    void midEventShouldGenerateDraftStandardDirectionDocument() throws Exception {
+        byte[] pdf = {1, 2, 3, 4, 5};
+        DocmosisDocument docmosisDocument = new DocmosisDocument("draft-standard-directions-order.pdf", pdf);
+        Document document = document();
+
+        given(documentGeneratorService.generateDocmosisDocument(any(), any()))
```

Comment:
Couldn't we be more specific than any()? We exactly know what document type should be generated, maybe we can ensure the map is passed properly as well?

Reply:
we could... but do I care about what docmosis was called with in this test? My assert is that a document exists

Reply:
Makes sense.